### ISSUE TITLE
Cannot assign to read only property 'month' of object '#<Object>'

### ISSUE THREAD

**Issue by vaynevayne** [mantinedev/mantine#4542]
### What package has an issue

@mantine/form

### Describe the bug

```jsx
<MonthPickerInput
              label={
                <Center inline>
                  消耗日期
                  <ActionIcon color="red" onClick={() => form.removeListItem('list', index)}>
                    <IconTrash size="1rem" />
                  </ActionIcon>
                </Center>
              }
              placeholder="选择消耗日期"
              valueFormat="YYYY-MM"
              {...form.getInputProps(`list.${index}.month`)}
              onChange={(value) => {
                form.getInputProps(`list.${index}.month`).onChange(value)
                console.log('onChange', value)

                const month = dayjs(value).startOf('month').format('YYYY-MM')
                console.log('month', month)
                form.setValues(
                  produce(form.values, (draft) => {
                    draft.list[index].dataSource = [
                      {
                        g_neo_channel: '渠道',
                        g_neo_supplier: '代理商',
                        cost_money: '12121',
                      },
                    ]
                  })
                )
              }}
            />
```
When you select the date, an error will be reported

### What version of @mantine/hooks page do you have in package.json?

latest

### Do you know how to fix the issue

None

### Are you willing to participate in fixing this issue and create a pull request with the fix

None

**Comment by vaynevayne** [comment]
<img width="653" alt="image" src="https://github.com/mantinedev/mantine/assets/124666577/876ad2f3-e9e4-4520-958b-5346c50d60e1">

I remember reporting the problem, and it was never fixed?

**Comment by vaynevayne** [comment]
useForm, since there is no performance optimization, such as responsive, it is better to use the syntactic sugar of useImmer + getInputProps, there is no need to implement it by yourself, there are many bugs, and it also limits the usage.

**Comment by rtivital** [comment]
Please send a sandbox that reproduces the issue, template –https://codesandbox.io/s/mantine-template-9ze89?file=/src/App.tsx:0-6

**Comment by vaynevayne** [comment]
The object returned by klona can not be modified, you have a problem with this code ah, error prompt is already obvious, brother

**Comment by Rigel772** [comment]
Similar problem here:
set-path.ts:10 Uncaught TypeError: Cannot assign to read only property 'name1' of object '#<Object>'
    at setPath (set-path.ts:10:27)
    at use-form.ts:96:22
    at basicStateReducer (react-dom.development.js:16540:41)
    at updateReducer (react-dom.development.js:16664:22)
    at updateState (react-dom.development.js:17004:10)
    at Object.useState (react-dom.development.js:17915:16)
    at useState (react.development.js:1622:21)
    at useForm (use-form.ts:41:32)
    at UserProfileForm (UserProfileForm.jsx:19:16)
    at renderWithHooks (react-dom.development.js:16305:18)

**Comment by ChrisLFieldsII** [comment]
Leaving this Fwiw, I ran into this error and spent ~30 mins debugging it.

Turns out I had a form component that could take an initial form values object and I passed this optional init form values into the `useForm.initialValues` hook.

The init form values object I was passing in was a react `useState` variable and that is what ultimately caused this "Cannot assign to read only property 'month' of object '#<Object>'" error.

I had to spread the `useState` variable into my form init values prop instead of passing it in directly and problem resolved
